aten/src/ATen/native/native_functions.yaml

```diff
@@ -2633,8 +2639,14 @@
 - func: neg_(Tensor(a!) self) -> Tensor(a!)
   use_c10_dispatcher: full
   variants: function, method
+  dispatch:
```

Comment:
Interesting... is this our usual pattern for adding sparse versions of functions, that only the inplace and out versions get an explicit sparse dispatch?

Reply:
@mruberry for an element-wise op, I'm not sure if there is a usual pattern for that as there are few functions support sparse. This is already done like that for 
- [log1p](https://github.com/pytorch/pytorch/blob/870f6470405a37392eb37ea7e927bca6adfecc3c/aten/src/ATen/native/native_functions.yaml#L1898)

If I understand correctly the dispatch mechanism, in case of `neg` it is routed to `neg_out` here : 
https://github.com/pytorch/pytorch/blob/88bc97465bbc8a2094db3695cf5cf99f14400ba4/aten/src/ATen/native/UnaryOps.cpp#L401
IMO, it is sufficient to set explicit sparse dispatch to inplace and out versions only. What do you think ?

Reply:
That's true in this case for forward, but this will cause sparse to inherit the dense autograd formulas these functions already have, right?

Reply:
Yes, that's a good point and for element-wise functions that has gradient of zero equal zero it can work. For others, like asin here, we can do similarly to `log1p_backward`:
https://github.com/pytorch/pytorch/blob/a475613d1d2cfef801231fe9004048fc5bf6b404/torch/csrc/autograd/FunctionsManual.cpp#L2576-L2580

Reply:
I guess this is OK. We should continue to think about how readable and maintainable this approach to dispatch is, however. It relies, for example, on these functions being implemented by calling their out variants. I don't think that's likely to change, but it could.